Sources/NIOTransportServices/NIOTSConnectionChannel.swift

```diff
@@ -349,6 +349,34 @@ extension NIOTSConnectionChannel: Channel {
     }
 
     public func getOption<Option: ChannelOption>(_ option: Option) -> EventLoopFuture<Option.Value> {
+        if #available(OSX 10.15, iOS 12.0, tvOS 12.0, watchOS 6.0, *) {
+            switch option {
+            case is NIOTSChannelOptions.Types.NIOTSEstablishmentReportOption:
+                let promise: EventLoopPromise<Option.Value> = eventLoop.makePromise()
+                do {
+                    try self.nwConnection0().requestEstablishmentReport(queue: self.connectionQueue) {
+                        promise.succeed($0 as! Option.Value)
+                    }
+                } catch {
+                    promise.fail(error)
+                }
+                return promise.futureResult
+                
+            case is NIOTSChannelOptions.Types.NIOTSDataTransferReportOption:
+                let promise: EventLoopPromise<Option.Value> = eventLoop.makePromise()
+                do {
+                    let pendingReport = try self.nwConnection0().startDataTransferReport()
+                    promise.succeed(pendingReport as! Option.Value)
+                } catch {
+                    promise.fail(error)
+                }
+                return promise.futureResult
+                
+            default:
+                break
+            }
+        }
```

Comment:
These implementations should be in `getOptions0`: the current implementation is not thread-safe.

Reply:
As `requestEstablishmentReport` is asynchronous, how should I handle returning the value via a future? `getOption` returns a future, so that's why I placed the option reading there. `getOptions0` should return the value directly.

Reply:
In this case, the return value of the option should itself be a future. The futures returned from `getOption` are primarily used for dealing with the fact that that interface needs to be thread-safe: it's fine to return a nested future here.